core/jvm/src/main/scala/fs2/interop/flow/package.scala

```diff
@@ -99,20 +202,69 @@ package object flow {
   ): Resource[F, Publisher[A]] =
     StreamUnicastPublisher(stream)
 
+  /** Allows subscribing a [[Subscriber]] to a [[Stream]].
+    *
+    * The returned program will run until
+    * all the stream elements were consumed.
+    * Cancelling this program will gracefully shutdown the subscription.
+    *
+    * @param stream the [[Stream]] that will be consumed by the subscriber.
+    * @param subscriber the [[Subscriber]] that will receive the elements of the stream.
+    */
+  def subscribeStream[F[_], A](
+      stream: Stream[F, A],
+      subscriber: Subscriber[A]
+  )(implicit
+      F: Async[F]
+  ): F[Unit] =
+    flow.toPublisher(stream).use { publisher =>
+      F.async_[Unit] { cb =>
```

Comment:
This `async_` is uncancelable (or will be, after the bug fix in CE 3.5.0 is release). We should use the `async` variant and return `Some(IO.unit)` so it is cancelable.

Reply:
IIRC the subscription process shouldn't be cancelable, but once it has been subscribed then we can cancel that `subscription`; which is done when the `Resource` is closed.